distribution/debian/scripts/postinst

```diff
@@ -1,21 +1,49 @@
 #!/bin/sh
 set -e
 
-# Add Vector to adm group to read /var/logs
-usermod --append --groups adm vector || true
-
-if getent group 'systemd-journal'
-then
-  # Add Vector to systemd-journal to read journald logs
-  usermod --append --groups systemd-journal vector || true
-  systemctl daemon-reload || true
-fi
-
-if getent group 'systemd-journal-remote'
-then
-  # Add Vector to systemd-journal-remote to read remote journald logs
-  usermod --append --groups systemd-journal-remote vector || true
-  systemctl daemon-reload || true
-fi
+create_user() {
+  id vector >/dev/null 2>&1 ||
+    useradd --system \
+      --shell /sbin/nologin \
+      --home-dir /var/lib/vector \
+      --user-group \
+      --comment "Vector observability data router" \
+      vector
+
+  # Add to adm group to read /var/logs
+  usermod --append --groups adm vector || true
+
+  # Add to systemd-journal to read journald logs
+  if getent group 'systemd-journal'; then
+    usermod --append --groups systemd-journal vector || true
+  fi
+
+  # Add to systemd-journal-remote to read remote journald logs
+  if getent group 'systemd-journal-remote'; then
+    usermod --append --groups systemd-journal-remote vector || true
+  fi
+}
+
+configure_permissions() {
+  if [ ! -d "/var/lib/vector" ]; then
+    # Create default Vector data directory
+    mkdir -p /var/lib/vector
+  fi
+
+  # Own and RW restrict Vector data directory
+  chown -R vector:vector /var/lib/vector
+  chmod 740 /var/lib/vector
+
+  # Own and RW restrict Vector config directory
+  chown -R vector:vector /etc/vector
+  chmod 740 /etc/vector
+}
+
+case "$1" in
+configure)
+  create_user
```

Comment:
This is making me realize we probably need a `postrm` script too to remove the user.

Reply:
https://wiki.debian.org/AccountHandlingInMaintainerScripts#:~:text=Account%20deletion,for%20example%20piuparts).

https://ma.ttias.be/on-removing-users-with-postun-in-rpm-spec-files/

This seems to be something up to debate, I opted for non-destructive.

Reply:
Hah, what mess. Yeah, I think removing the user on uninstall is out of scope anyway, so we can leave the existing behavior for now. I can see the arguments for both.